packages/shadydom/src/patches/Node.js

```diff
@@ -308,7 +308,33 @@ export const NodePatches = utils.getOwnPropertyDescriptors({
         !currentScopeIsCorrect(node, newScopeName);
     const needsSlotFinding = ownerRoot && !node['__noInsertionPoint'] &&
         (!preferPerformance || node.nodeType === Node.DOCUMENT_FRAGMENT_NODE);
-    if (needsSlotFinding || needsScoping) {
+
```

Comment:
This function is getting unwieldy. Think it'd improve readability to factor all the `needs-*` flags into separate functions and move the comments there.